pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -79,7 +82,21 @@ int getPriority(BaseFilterOperator filterOperator) {
           return 3;
         }
         if (filterOperator instanceof ScanBasedFilterOperator) {
-          return 4;
+          DataSourceMetadata metadata = filterOperator.getDataSourceMetadata();
+          int basePriority = 4;
+          if (metadata == null) {
+            // shouldn't happen
+            return basePriority;
+          }
+          // lower priority for multivalue columns
+          if (! metadata.isSingleValue()) {
+            // ensure priority for multivalue cols is higher than singvalue
+            // (accounting for the cardinality for single value)
+            basePriority *= DEFAULT_MAX_CARDINALITY;
```

Comment:
How did we decide upon the value for DEFAULT_MAX_CARDINALITY?

Reply:
Its a guesstimate at this point - let me know if we can use a better value or a better way to determine the value.